src/main/java/org/recordrobotics/charger/commands/manual/ManualClaw.java

```diff
@@ -0,0 +1,71 @@
+package org.recordrobotics.charger.commands.manual;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import org.recordrobotics.charger.subsystems.Claw;
+import org.recordrobotics.charger.control.IControlInput;
+
+public class ManualClaw extends CommandBase {
+
+	private Claw _claw;
+	private IControlInput _controls;
+
+	private static final double TURN_SPEED = 0.3;
+	// TODO: find a good value experimentally
+	private static final double ERROR_MARGIN = 0.01;
+
+	public ManualClaw(Claw claw, IControlInput controls) {
+		if (claw == null) {
```

Comment:
There's a difference between : "I assume this thing will not be null. if so, it's a programming error" vs "I can't assume this thing will not be null due to some condition I can't control (like resource contention or user actions)"
The 1st, you should use `assert`, like `assert claw != null : "Claw is null";`  -- simpler code + communicate that this is an assumption.
The 2nd one, yes, use if statement and throw.

is there reasons why you guys don't use assert ?

Reply:
The reason I chose not to use assertions is because they are disabled by default in Java and it's very inconvenient to modify the run script on the NI.

Reply:
👍